pkg/policy/ebpf.go

```diff
@@ -0,0 +1,726 @@
+package policy
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"os"
+	"strconv"
+	"unsafe"
+
+	bpf "github.com/aquasecurity/libbpfgo"
+
+	"github.com/aquasecurity/tracee/pkg/bufferdecoder"
+	"github.com/aquasecurity/tracee/pkg/containers"
+	"github.com/aquasecurity/tracee/pkg/errfmt"
+	"github.com/aquasecurity/tracee/pkg/events"
+	"github.com/aquasecurity/tracee/pkg/filters"
+	"github.com/aquasecurity/tracee/pkg/logger"
+	"github.com/aquasecurity/tracee/pkg/utils"
+	"github.com/aquasecurity/tracee/pkg/utils/proc"
+)
+
+const (
+	// outer maps
+	UIDFilterMapVersion         = "uid_filter_version"
+	PIDFilterMapVersion         = "pid_filter_version"
+	MntNSFilterMapVersion       = "mnt_ns_filter_version"
+	PidNSFilterMapVersion       = "pid_ns_filter_version"
+	UTSFilterMapVersion         = "uts_ns_filter_version"
+	CommFilterMapVersion        = "comm_filter_version"
+	CgroupIdFilterVersion       = "cgroup_id_filter_version"
+	ProcessTreeFilterMapVersion = "process_tree_map_version"
+	BinaryFilterMapVersion      = "binary_filter_version"
+
+	// inner maps
+	UIDFilterMap         = "uid_filter"
+	PIDFilterMap         = "pid_filter"
+	MntNSFilterMap       = "mnt_ns_filter"
+	PidNSFilterMap       = "pid_ns_filter"
+	UTSFilterMap         = "uts_ns_filter"
+	CommFilterMap        = "comm_filter"
+	CgroupIdFilterMap    = "cgroup_id_filter"
+	ProcessTreeFilterMap = "process_tree_map"
+	BinaryFilterMap      = "binary_filter"
+
+	ProcInfoMap = "proc_info_map"
+)
+
+// equality mirrors the C eq_t struct
+type equality struct {
+	equalInPolicies       uint64
+	equalitySetInPolicies uint64
+}
+
+const (
+	// 8 bytes for equalInPolicies and 8 bytes for equalitySetInPolicies
+	equalityValueSize = 16
+)
+
+type filtersEqualities struct {
+	uidEqualities      map[uint64]equality
+	pidEqualities      map[uint64]equality
+	mntNSEqualities    map[uint64]equality
+	pidNSEqualities    map[uint64]equality
+	cgroupIdEqualities map[uint64]equality
+	utsEqualities      map[string]equality
+	commEqualities     map[string]equality
+	procTreeEqualities map[uint64]equality
+	binaryEqualities   map[filters.NSBinary]equality
+}
+
+// TODO: refactor this method deduplicating code
+func (ps *Policies) computeFilterEqualities(fEqs *filtersEqualities, cts *containers.Containers) error {
```

Comment:
Needs documentation. Very obscure for new comers to the project. Consider 2 or 3 new engineers arriving into the project and having to read/maintain this. This should be the mindset when documenting (IMO).